content/how-to/bpmn/use-jsonata.md

```diff
@@ -0,0 +1,772 @@
+Parameters that accept expressions include:
+
+- **API payloads**. Calls to the Rhize GraphQL API and to external APIs can both use JSONata in their payloads.
+- **Outgoing payloads**. The messages you send with intermediate message throws can have JSONata.
+- **Response transformations.**
+  - REST. Write JSONata in the input expression to filter data. This is helpful to remove data from over-fetching and to keep the process variable size lower.
+  - GraphQL input. As with JSONata, your GraphQL tasks can pass their results to JSONata. This is less common, as GraphQL already provides tight control of the data it returns, but it is useful for calculating derived values.
+
+Review the full list of parameters that accept JSONata in the [BPMN element reference]({{< relref "bpmn-elements" >}}).
+
+### JSONata version
+
+Many implementations of JSONata exist.
+Rhize uses a custom Go implementation for high performance and safe calculation.
+
+## JSONata examples
+
+These snippets provide some examples of JSONata from manufacturing workflows.
+To experiment with how they work, copy the data and expression into a [JSONata exerciser](https://try.jsonata.org/) and try changing values.
+
+### Filter for items that contain
+
+This expression returns the ID of all `equipmentActual` items that are associated with a specified job response `JR-4`.
+It outputs the IDs as an array of strings in a new custom object.
+
+This is a minimal example of how you can use JSONata to transform data into new representations.
+Such transformation is a common prerequisite step for post-processing and service interoperability.
+```
+$.data.queryJobResponse[`id`="JR-4"].(
+    {"associatedEquipment": equipmentActual.id}
+)
+```
+
+{{% tabs %}}
+
+{{% tab "Input" %}}
+```json
+{
+  "data": {
+    "queryJobResponse": [
+      {
+        "id": "JR-1",
+        "data": [
+          {
+            "value": 100
+          }
+        ],
+        "equipmentActual": [
+          {
+            "id": "hauler"
+          },
+          {
+            "id": "actuator-121"
+          }
+        ]
+      },
+      {
+        "id": "JR-4",
+        "data": [
+          {
+            "value": "101.8"
+          }
+        ],
+        "equipmentActual": [
+          {
+            "id": "actuator-132"
+          },
+          {
+            "id": "actuator-133"
+          }
+        ]
+      }
+    ]
+  }
+}
+```
+
+{{% /tab %}}
+
+{{% tab "Output " %}}
+
+```json
+{
+  "associatedEquipment": [
+    "actuator-132",
+    "actuator-133"
+  ]
+}
+```
+{{% /tab %}}
+{{% /tabs %}}
+
+### Find actual associated with high values
+
+This expression finds all job responses whose `value` exceeds `100`.
+It outputs the matching job response IDs along with the associated equipment actual used in the job.
+
+In production, you may use a similar analysis to isolate all {{< abbr "resource actual" >}}s associated with an abnormal production outcome.
+
+```
+$map($.data.queryJobResponse, function($v,$i,$a){
```

Comment:
$i and $a are not used. Its probably best practice not to use more function signatures than needed
```
$map($.data.queryJobResponse, function($v){
    $number($v.data.value) > 102
        ?  {"jobResponseId": $v.id, "EquipmentActual": $v.equipmentActual}
        }
    )